test/pytest/service-desktop/test_messages.py

```diff
@@ -1,203 +1,376 @@
+    assert result, "Failed to get message by id!"
+
+    result, response = messages_tester.get_messages_by_thread_id(
+        received_message_1_record["threadID"], 0, 0)
+    assert result, "Failed to get messages by thread id!"
+    found_message_1_id = False
+    for record in response.messages:
+        if message_1_record["messageID"] == record["messageID"]:
+            found_message_1_id = True
+            break
+    found_message_2_id = False
+    for record in response.messages:
+        if message_2_record["messageID"] == record["messageID"]:
+            found_message_2_id = True
+            break
+    assert found_message_1_id and found_message_2_id
+
+    assert messages_tester.delete_message_by_id(message_1_record["messageID"]), "Failed to delete a message!"
+    assert messages_tester.delete_message_by_id(message_2_record["messageID"]), "Failed to delete a message!"
+
+    result, received_messages_records_count = messages_tester.get_messages_count()
+    assert result, "Failed to get messages count!"
+    assert received_messages_records_count == initial_number_of_messages_records, "Wrong number of messages!"
 
 
 @pytest.mark.service_desktop_test
 @pytest.mark.usefixtures("phone_unlocked")
-def test_remove_message(harness):
-    # get messages count
-    body = {"category": "message", "count": True}
-    ret = harness.endpoint_request("messages", "get", body)
-    assert ret["status"] == status["OK"]
-
-    count = ret["body"]["count"]
-    assert count > 0
-
-    # get message
-    number_of_requested_messages = 1
-    body = {"category": "message", "limit": number_of_requested_messages, "offset": 0}
-    ret = harness.endpoint_request("messages", "get", body)
-    assert ret["status"] == status["OK"]
-
-    messages = ret["body"]["entries"]
-    messages_count = len(messages)
-    assert messages_count == number_of_requested_messages
-
-    # remove message
-    sms_to_remove = messages[0]
-    body = {"category": "message", "messageID": sms_to_remove["messageID"]}
-    ret = harness.endpoint_request("messages", "del", body)
-    assert ret["status"] == status["NoContent"]
-
-    # get messages count again
-    body = {"category": "message", "count": True}
-    ret = harness.endpoint_request("messages", "get", body)
-    assert ret["status"] == status["OK"]
-    assert ret["body"]["count"] == count - 1
+def test_get_multiple_messages(harness):
+    messages_tester = MessagesTester(harness)
+    result, received_messages_records_count = messages_tester.get_messages_count()
+    assert result, "Failed to get messages count!"
+    initial_number_of_messages_records = received_messages_records_count
+
+    message_number_1 = "123456789"
+    message_1_body = "Hello, how are you?"
+    message_2_body = "Are you there?"
+
+    message_number_2 = "223456789"
+    message_3_body = "What's up?"
+    message_4_body = "Did you checked your email recently?"
+
```

Comment:
instead of copying why not create for loop? same below

Reply:
Added some loops